handle get_attr opearations in typechecker

migeed-z opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#62682**
* #62582

Differential Revision: [D30107789](https://our.internmc.facebook.com/intern/diff/D30107789)

migeed-z committed:
handle get_attr opearations in typechecker

[ghstack-poisoned]

migeed-z commented:
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30107789).

jamesr66a commented on torch/fx/experimental/graph_gradual_typechecker.py:
```diff
@@ -508,8 +508,17 @@ def type_check_node(self, n: Node):
+
         elif n.op == 'call_function':
-            if n.target in _INFERENCE_RULES:
+            if n.target in _INFERENCE_RULES and n.target == getattr:
```
Is `getattr` always going to be in `_INFERENCE_RULES`? If so, could just make this condition `if n.target == getattr` and then `assert getattr in _INFERENCE RULES` the branch. That would simplify reading the logic here.

migeed-z committed:
Update on "handle get_attr opearations in typechecker"


Differential Revision: [D30107789](https://our.internmc.facebook.com/intern/diff/D30107789)

[ghstack-poisoned]

migeed-z committed:
Update on "handle get_attr opearations in typechecker"


Differential Revision: [D30107789](https://our.internmc.facebook.com/intern/diff/D30107789)

[ghstack-poisoned]

migeed-z commented:
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30107789).
